internal/worker.go

```diff
@@ -49,14 +49,27 @@ type (
 		// default: defaultMaxConcurrentActivityExecutionSize(1k)
 		MaxConcurrentActivityExecutionSize int
 
-		// Optional: Sets the rate limiting on number of activities that can be executed per second. Notice that the
-		// number is represented in float, so that you can set it to less than 1 if needed. For example, set the number
-		// to 0.1 means you want your activity to be executed once for every 10 seconds. This can be used to protect
-		// down stream services from flooding.
+		// Optional: Sets the rate limiting on number of activities that can be executed per second per
+		// worker. Notice that the number is represented in float, so that you can set it to less than
+		// 1 if needed. For example, set the number to 0.1 means you want your activity to be executed
+		// once for every 10 seconds. This can be used to protect down stream services from flooding.
 		// The zero value of this uses the default value.
 		// default: defaultMaxActivityExecutionRate(100k)
-		// Warning: activity's StartToCloseTimeout starts ticking even if a task is blocked due to rate limiting.
-		MaxActivityExecutionPerSecond float64
+		// Warning: activity's StartToCloseTimeout starts ticking even if a task is blocked due to
+		// rate limiting.
+		WorkerActivitiesPerSecond float64
+
+		// Optional: Sets the rate limiting on number of activities that can be executed per second.
+		// This is managed by the server and controls activities per second for your entire tasklist
+		// whereas WorkerActivityTasksPerSecond controls activities only per worker.
+		// Notice that the number is represented in float, so that you can set it to less than
+		// 1 if needed. For example, set the number to 0.1 means you want your activity to be executed
+		// once for every 10 seconds. This can be used to protect down stream services from flooding.
+		// The zero value of this uses the default value.
+		// default: defaultMaxActivityExecutionRate(100k)
+		// Warning: activity's StartToCloseTimeout starts ticking even if a task is blocked due to
```

Comment:
I don't think it is true for global throttling.

Reply:
If throttling kicks in, the task will not be delivered to the poll channel and the activity's StartToCloseTimeout starts ticking till the poll times out.

Reply:
It will actually be the *ScheduleToStart timeout in this case I think. Since the task is in there, but fails to get picked up by the worker since throttling is in effect.

Reply:
It is what I mean. We don't need this warning as activity just stays queued in the task list.